src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -518,6 +518,18 @@ class Manager : public DoutPrefixProvider {
           }
         }
       }
+
+      // updating perfcounters with topic stats
+      uint64_t entries_size;
+      uint32_t entries_number;
+      const auto ret = cls_2pc_queue_get_topic_stats(rados_ioctx, queue_name, entries_number, entries_size);
+      if (ret < 0) {
+        ldpp_dout(this, 1) << "ERROR: topic stats for topic: " << queue_name << ". error: " << ret << dendl;
+      }
+
+      auto counters = rgw::persistent_topic_counters::get(queue_name);
+      rgw::persistent_topic_counters::set(counters, l_rgw_persistent_topic_len, entries_number);
```

Comment:
should we add also the reservations.size() here ? need to get the cls_2pc_queue_list_reservations() for topic and add the size the queue_length?

Reply:
@yuvalif what do you think?

Reply:
"reservations" is just counting the amount of entries that are reserved but not comitted (pushed) yet, not the number of entries in the queue.
this number would be close to zero for most of the time, and may increase only if an RGW crashed, and there are reservations that were not comitted (that will eventually expire).
I don't think we should expose that to something like prometheus, and keep that only for case of debugging after a crash, using radosgw-admin.